### PR TITLE
Fix item create behavior for exploration and downtime actions

### PR DESCRIPTION
**Pull request by CarlosFdez**
Also moves us away from ramming dataset into item source and hoping foundry can figure it out.

### CODE REVIEW

File: src/module/actor/sheet/helpers.ts [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
+        type: "spell",
+        name: R.compact([newLabel, levelLabel, spellLabel]).join(" "),
+        system: {
+            level: { value: level || 1 },
+            location: { value: String(data.location) },
+            traits: {
+                value: level === 0 ? ["cantrip"] : [],
+            },
+        },
+    } satisfies DeepPartial<SpellSource>;
```
Comment (stwlam) [inline src/module/actor/sheet/helpers.ts:31]: Any reason to not just do a regular type annotation here?

Comment (CarlosFdez) [reply src/module/actor/sheet/helpers.ts:31]: the mergeObject() later doesn't work if I do a regular type annotation, it doesn't know that "system" as a property exists.

Comment (stwlam) [reply src/module/actor/sheet/helpers.ts:31]: but that's exactly what's happened below

Comment (CarlosFdez) [reply src/module/actor/sheet/helpers.ts:31]: The `?? {}` is the part that's different.

Comment (stwlam) [reply src/module/actor/sheet/helpers.ts:31]: Ohh, I see.

Comment (stwlam) [reply src/module/actor/sheet/helpers.ts:31]: buenooooo

**Pull request merged**